### DISCUSSION TITLE
Having trouble getting FCM token iOS

### DISCUSSION THREAD

**Discussion by kellymmcd** [shinyorg/shiny#1513] [Q&A]
Trying to set up Maui app to receive firebase notifications. I'm using the Shiny.Push.FirebaseMessaging  nuget package 4.0.0 beta and having an issue in iOS where the app is hanging when calling RequestAccess()

I have downloaded pushtester app from https://github.com/shinyorg/pushtester updated this to use 4.0.0 beta and then followed the setup instructions for my app but it still hangs when calling RequestAccess after trying to register. 

My iOS app is using the same bundle id as I have set up in firebase and I have uploaded a APNs Authentication key to firebase project.
I am testing on a physical device, have tried in debug and release.

Does anyone know what could be causing this to hang for me? Any help would be greatly appreciated.

**Comment by munkii** [comment]
Have you visited the Shiny AppBuilder to get the necessary wireup? https://shinylib.net/client/appbuilder/

My guess is that you have not wired up the callbacks in your AppDelegate.

**Comment by aritchie** [comment]
@kellymmcd It is a beta and an initial beta at that.  Support is not offered on betas.

**Comment by PXInformatica** [comment]
Hello, Did you manage to solve the problem? Now I find myself in the same situation.

**Reply by munkii** [reply]
And you have wired up the aforementioned callbacks? 

Been to appbuilder?
